microcloud/cmd/microcloud/main_init.go

```diff
@@ -139,11 +139,21 @@ func (c *cmdInit) Run(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func lookupPeers(s *service.Handler, autoSetup bool, subnet *net.IPNet, systems map[string]InitSystem) error {
+// lookupPeers attempts to find eligible systems over mDNS, optionally limiting lookup to the given subnet if not nil.
+// Found systems will be progressively added to a table, and the user selection is added to the `systems` map.
+//
+// - If `autoSetup` is true, all systems found in the first 5s will be recorded, and no other input is required.
+// - `expectedSystems` is a list of expected hostnames. If given, the behaviour is similar to `autoSetup`,
+// except it will wait up to a minute for exclusively these systems to be recorded.
+func lookupPeers(s *service.Handler, autoSetup bool, subnet *net.IPNet, expectedSystems []string, systems map[string]InitSystem) error {
```

Comment:
Thanks for the comment, this explains it well.

This isn't a blocker, just a nit, but the coupling of different timeouts with different modes feels quite peculiar for a utility function like this.

For future reference pushing the timeout decision to the caller and just having a ctx or timeout argument here would make the function description cleaner to understand and separate the concerns of what the function should do by when.